src/ramptr.c

```diff
@@ -0,0 +1,23 @@
+#include "global.h"
+#include "ramptr.h"
+#include "data.h"
+#include "functions.h"
+
+void sub_08012C44(void)
+{
+    CpuFill16(0, gUnk_03006590, 0x218);
+    gUnk_030067B0 = (u32*)EWRAM_START;
+    gUnk_0300657C = gUnk_02000100;
+    gUnk_03006584 = gUnk_02000380;
+    gUnk_030067AC = gUnk_02000540;
+    gUnk_03006570 = gUnk_02000580;
+    gUnk_030067A8 = gUnk_02000680;
+    CpuFill32(0, (u32*)EWRAM_START, 0x400);
```

Comment:
Not sure what happens here. 0x2000000, 0x2000100, 0x2000380 are referenced directly but here the initialization is overlapping all of them. This is very bad as in C the memory layout is not guaranteed (it is specified in the linker script but that's out of the language itself) so this can be out-of-bounds write. 
(Not asking you to change anything -- I'm just pointing out the problem)